Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/Cancellation/CancelProgressFlowNode.cs

```diff
@@ -58,6 +58,14 @@ public override string ToString()
 
         private JobHandle ScheduleCheckCancelProgressJob(JobHandle dependsOn)
         {
+            if (!m_ProgressLookupData.IsDataInvalidated && m_ParentProgressLookupData is
+            {
+                IsDataInvalidated: false
+            })
```

Comment:
just to double check, this is just syntax sugar. It doesn't resolve to some sort of reflection.

Reply:
Yeah it's just Rider merging the following into a pattern.

`if (!m_ProgressLookupData.IsDataInvalidated && (m_ParentProgressLookupData != null && !m_ParentProgressLookupData.IsDataInvalidated))`

I've never seen it do it like that before but there isn't any reflection or allocations going on.

Reply:
OK, just wanted to double check because I have seen that pattern matching style produce less efficient code